prov/shm/src/smr_ep.c

```diff
@@ -382,18 +382,18 @@ static struct fi_ops smr_ep_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
-static int smr_endpoint_name(char *name, char *addr, size_t addrlen,
-			     int pid, int dom_idx, int ep_idx)
+static void smr_endpoint_name(char *name, char *addr, size_t addrlen,
+			      int dom_idx, int ep_idx)
 {
+	char *start;
 	memset(name, 0, SMR_NAME_SIZE);
-	if (addr) {
```

Comment:
The change loses this check against a null addr.  Do all callers provide a non-null address?

Reply:
The addr comes from info->src_addr which is guaranteed to be non-null after getinfo with the correct address. Unless the user frees src_addr, it should be ok to drop the check, but it might be safer to keep it after all.